include/nbl/video/IGPUCommandBuffer.h

```diff
@@ -1,264 +1,81 @@
-        uint32_t bufferMemoryBarrierCount, const VkBufferMemoryBarrier* pBufferMemoryBarriers,
-        uint32_t imageMemoryBarrierCount, const VkImageMemoryBarrier* pImageMemoryBarriers) = 0;*/
-
-    //virtual void beginRenderPass(const VkRenderPassBeginInfo* pRenderPassBegin, VkSubpassContents content) = 0;
-    //virtual void nextSubpass(VkSubpassContents contents) = 0;
-    //virtual void endRenderPass() = 0;
-
-    virtual void setDeviceMask(uint32_t deviceMask) { m_deviceMask = deviceMask; }
-
-    //those two instead of bindPipeline(E_PIPELINE_BIND_POINT, pipeline)
-    virtual void bindGraphicsPipeline(IGPURenderpassIndependentPipeline* pipeline) = 0;
-    virtual void bindComputePipeline(IGPUComputePipeline* pipeline) = 0;
-
-    //virtual void resetQueryPool(IGPUQueryPool* queryPool, uint32_t firstQuery, uint32_t queryCount) = 0;
-    //virtual void beginQuery(IGPUQueryPool* queryPool, uint32_t entry, std::underlying_type_t<E_QUERY_CONTROL_FLAGS> flags) = 0;
-    //virtual void endQuery(IGPUQueryPool* queryPool, uint32_t query) = 0;
-    //virtual void copyQueryPoolResults(IGPUQueryPool* queryPool, uint32_t firstQuery, uint32_t queryCount, IGPUBuffer* dstBuffer, size_t dstOffset, size_t stride, std::underlying_type_t<E_QUERY_RESULT_FLAGS> flags) = 0;
-    //virtual void writeTimestamp(std::underlying_type_t<E_PIPELINE_STAGE_FLAGS> pipelineStage, IGPUQueryPool* queryPool, uint32_t query) = 0;
-
-    virtual void bindDescriptorSets(E_PIPELINE_BIND_POINT pipelineBindPoint, IGPUPipelineLayout* layout, uint32_t firstSet, uint32_t descriptorSetCount,
-        IGPUDescriptorSet** pDescriptorSets, uint32_t dynamicOffsetCount, const uint32_t pDynamicOffsets
-    ) = 0;
-    virtual void pushConstants(IGPUPipelineLayout* layout, std::underlying_type_t<IGPUSpecializedShader::E_SHADER_STAGE> stageFlags, uint32_t offset, uint32_t size, const void* pValues) = 0;
-
-    //virtual void clearColorImage(IGPUImage* image, IGPUImage::E_LAYOUT imageLayout, const VkClearColorValue* pColor, uint32_t rangeCount, const VkImageSubresourceRange* pRanges) = 0;
-    //virtual void clearDepthStencilImage(IGPUImage* image, IGPUImage::E_LAYOUT imageLayout, const VkClearDepthStencilValue* pDepthStencil, uint32_t rangeCount, const VkImageSubresourceRange* pRanges) = 0;
-    //virtual void clearAttachments(uint32_t attachmentCount, const VkClearAttachment* pAttachments, uint32_t rectCount, const VkClearRect* pRects) = 0;
-    virtual void fillBuffer(IGPUBuffer* dstBuffer, size_t dstOffset, size_t size, uint32_t data) = 0;
-    virtual void updateBuffer(IGPUBuffer* dstBuffer, size_t dstOffset, size_t dataSize, const void* pData) = 0;
-
-protected:
-    //! `_flags` takes bits from E_USAGE
-    virtual void begin(uint32_t _flags)
+    static void bindDescriptorSets_generic(const IGPUPipelineLayout* _newLayout, uint32_t _first, uint32_t _count, const IGPUDescriptorSet* const* _descSets, const IGPUPipelineLayout** _destPplnLayouts)
```

Comment:
just make sure to note that we dont give a fuck about the push constant ranges (unlike Vulkan, and that will require special emulation when we implement the Vulkan cmdbuffer)

Reply:
we dont